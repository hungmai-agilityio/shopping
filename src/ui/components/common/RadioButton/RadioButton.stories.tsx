import { Meta, StoryObj } from '@storybook/react';

import RadioButton from '@/ui/components/common/RadioButton';

const meta: Meta<typeof RadioButton> = {
  component: RadioButton,
  tags: ['autodocs'],
  parameters: {
    layout: 'centered'
  },
  argTypes: {
    checked: {
      description: 'Default value checked'
    },
    id: {
      description: 'Id of the radio station'
    },
    label: {
      description: 'Radio label'
    },
    value: {
      description: 'Set value of radio'
    },
    name: {
      description: 'Radio name'
    },
    onChange: {
      description: 'Handle change radio active'
    }
  }
};

export default meta;

type Story = StoryObj<typeof RadioButton>;

export const Default: Story = {
  args: {
    checked: true,
    id: 'male',
    label: 'Male',
    name: 'gender'
  }
};
