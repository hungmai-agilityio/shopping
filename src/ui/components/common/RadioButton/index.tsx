import { ChangeEvent, memo } from 'react';

// Constants
import { popping } from '@/constants';

interface RadioProps {
  id: string;
  value: string;
  label: string;
  name: string;
  checked: boolean;
  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
}

const RadioButton = memo(
  ({ id, value, label, name, checked, onChange }: RadioProps) => {
    return (
      <label
        htmlFor={id}
        className={`${popping.className} flex items-center font-bold text-xs cursor-pointer`}
      >
        <input
          id={id}
          type="radio"
          name={name}
          value={value}
          checked={checked}
          className="accent-second h-4 w-4 cursor-pointer mr-2"
          onChange={onChange}
        />
        {label}
      </label>
    );
  }
);

export default RadioButton;
