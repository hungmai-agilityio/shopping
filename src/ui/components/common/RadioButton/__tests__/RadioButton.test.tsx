import { fireEvent, render, screen } from '@testing-library/react';

// Components
import RadioButton from '@/ui/components/common/RadioButton';

describe('RadioButton component', () => {
  const defaultProps = {
    id: '1',
    checked: false,
    label: 'radio button',
    name: '',
    value: '',
    onChange: jest.fn()
  };

  it('Should be render with checked', () => {
    render(<RadioButton {...defaultProps} checked />);

    const checkbox = screen.getByRole('radio') as HTMLInputElement;
    expect(checkbox.checked).toBe(true);
  });

  it('should be render match to snapshot', () => {
    const { container } = render(<RadioButton {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });
});
