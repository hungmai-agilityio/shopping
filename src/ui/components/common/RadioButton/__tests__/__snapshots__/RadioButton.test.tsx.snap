// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton component should be render match to snapshot 1`] = `
<div>
  <label
    class="className flex items-center font-bold text-xs cursor-pointer"
    for="1"
  >
    <input
      class="accent-second h-4 w-4 cursor-pointer mr-2"
      id="1"
      name=""
      type="radio"
      value=""
    />
    radio button
  </label>
</div>
`;
